parsing/lexer.mll

```diff
@@ -386,11 +383,15 @@ rule token = parse
   | "\'" newline "\'"
       { update_loc lexbuf None 1 false 1;
         CHAR (Lexing.lexeme_char lexbuf 1) }
-  | "\'" [^ '\\' '\'' '\010' '\013'] "\'"
+  | "\'" [^ '\\' '\'' '\010' '\013' '\x80'-'\xff'] "\'"
       { CHAR(Lexing.lexeme_char lexbuf 1) }
+  | "\'" ['\x80'-'\xff'] "\'"
+      { raise (Error(Illegal_character (Lexing.lexeme_char lexbuf 1),
+                     Location.curr lexbuf))
+      }
   | "\'\\" ['\\' '\'' '\"' 'n' 't' 'b' 'r' ' '] "\'"
       { CHAR(char_for_backslash (Lexing.lexeme_char lexbuf 2)) }
-  | "\'\\" ['0'-'9'] ['0'-'9'] ['0'-'9'] "\'"
+  | "\'\\" ['0'-'2'] ['0'-'9'] ['0'-'9'] "\'"
```

Comment:
Why is this here?

Reply:
Ah. I can remove it, but I noticed most of the range for character decimal escapes was not going to parse anyway. (The function that converts it to a char will notice it's above 255 but it isn't necessary to accept, say, 300 or above anyway.)

Reply:
I think it changes the exception, though.